repo/blob/s3/s3_storage.go

```diff
@@ -312,6 +312,40 @@ func (s *s3Storage) FlushCaches(ctx context.Context) error {
 	return nil
 }
 
+func isAccessDenied(err error) bool {
+	var e minio.ErrorResponse
+	return errors.As(err, &e) && e.Code == "AccessDenied"
+}
+
+func (s *s3Storage) bucketExists(ctx context.Context) (bool, error) {
+	ok, err := s.cli.BucketExists(ctx, s.Options.BucketName)
+	if err == nil {
+		return ok, nil
+	}
+
+	if !isAccessDenied(err) {
+		return false, errors.Wrap(err, "error checking bucket existence")
+	}
+
+	if s.Options.Prefix == "" {
+		return false, errors.Wrap(err, "no permission to access bucket itself")
+	}
+
+	// Verify the existence of bucket by getting an non-existent object from the bucket/prefix and
+	// expecting an BlobNotFound error, in case that the account has no permission to the bucket itself
+	// but has permission to the prefix of the bucket
+	nonExistentBlob := fmt.Sprintf("kopia-s3-storage-initializing-%v", clock.Now().UnixNano())
+
+	var scOutput gather.WriteBuffer
```

Comment:
add `defer scOutput.Close()` and while at it rename it to something like `out`

Reply:
Done the changes. Thanks

Reply:
I do not see the changes (yet).